### DISCUSSION TITLE
What are the benefits of using QMP for qemu administration instead of using libvirt?

### DISCUSSION THREAD

**Discussion by alishir** [kata-containers/kata-containers#1871] [Q&A] [answered]
I just read [this blog post ](https://www.redhat.com/en/blog/all-you-need-know-about-kvm-userspace) and realized that kata-containers manages qemu instances directly via QMP.
I'm curious about this architectural decision, while many projects are related to the management of qemu instances using libvirt.
Would you please explain the advantages/disadvantages of using qmp instead of libvirt?

**Comment by devimc** [comment, marked as answer]
Hi @alishir 

we took this decision long time ago (~6 years) in the Clear Containers project, I don't remember all the points:
* we can't depend on a daemon running in the host
* Dependencies: libvirt is another package than must be installed in the host
* Performance: govmm is smaller and may consume less RAM than libvirt-go
* Security (I don't remember very well this point): Anyone with access to the libvirt socket could modify any VM

**Reply by alishir** [reply]
Hi @devimc 

Thanks so much for your detailed explanation.
